pkg/servicemonitor/servicemonitor_test.go

```diff
@@ -1,207 +1,193 @@
-package servicemonitor_test
+package servicemonitor
 
 import (
-	"github.com/golang/mock/gomock"
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-
 	"context"
-
-	// tested package
+	"testing"
 
 	"github.com/openshift/route-monitor-operator/api/v1alpha1"
-	consterror "github.com/openshift/route-monitor-operator/pkg/consts/test/error"
-	constinit "github.com/openshift/route-monitor-operator/pkg/consts/test/init"
-	"github.com/openshift/route-monitor-operator/pkg/servicemonitor"
-
-	clientmocks "github.com/openshift/route-monitor-operator/pkg/util/test/generated/mocks/client"
-	utilmock "github.com/openshift/route-monitor-operator/pkg/util/test/generated/mocks/reconcile"
-	testhelper "github.com/openshift/route-monitor-operator/pkg/util/test/helper"
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	rhobsv1 "github.com/rhobs/obo-prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
-type ResourceComparerMockHelper struct {
-	CalledTimes int
-	ReturnValue bool
+func TestTemplateAndUpdateServiceMonitorDeployment(t *testing.T) {
```

Comment:
+1 to simplify unit tests. However, the rest of the code base uses BDD tests with Gomega assertions. Why do you think this specific file would benefit from table tests using `assert.` assertions? Should we use Ginkgo table tests instead?